modules/qtwidgets/src/rangesliderqt.cpp

```diff
@@ -260,14 +274,16 @@ void RangeSliderQt::moveMiddle(int delta) {
         QSplitter::setSizes(sizes);
     }
     // update internal state
-    updateSplitterPosition(0, 0);
+    updateStateFromSliders();
+    emit valuesChanged(value_.x, value_.y);
+
     parentWidget()->repaint();
     repaint();
 }
 
 bool RangeSliderQt::eventFilter(QObject* obj, QEvent* event) {
     if (event->type() == QEvent::ToolTip) {
-        QHelpEvent* helpEvent = static_cast<QHelpEvent*>(event);
+        const auto* helpEvent = dynamic_cast<QHelpEvent*>(event);
```

Comment:
I know that ClangTidy thinks one should use dynamic casts here. But Im not sure we should follow. We to check the event type(), so we should know the object type and static casting should be fine. And since this might trigger a lot... dynamic cast might be a bit more expensive? Not that I have tested that in the last 5+ years...

Reply:
I'll revert the changes.